### PR TITLE
HOSTEDCP-1442: Split AWS security groups for workers and for VPC endpoint

### PR DESCRIPTION
**Pull request by csrwng**
**What this PR does / why we need it**:
Before this commit, a single security group was used for both workers and VPC endpoint. Now we create a separate security group for each and continually reconcile them.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #[HOSTEDCP-1442](https://issues.redhat.com/browse/HOSTEDCP-1442)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.

### CODE REVIEW

**Comment by Patryk-Stefanski** [comment]
/retitle HOSTEDCP-1442: Split AWS security groups for workers and for VPC endpoint

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go [hunk 1]
```diff
@@ -4498,187 +4469,285 @@ func healthCheckIdentityProvider(ctx context.Context, hcp *hyperv1.HostedControl
...
+	if shouldReconcileVPCEndpointSG {
+		vpcEndpointSG, err = existingSG(ctx, r.ec2Client, awsVPCEndpointSecurityGroupName(hcp.Spec.InfraID), hcp.Spec.InfraID)
+		if err != nil {
+			return "", "", err
+		}
 	}
-	vpcCIDR := awssdk.StringValue(vpcResult.Vpcs[0].CidrBlock)
-	describeSGResult, err := ec2Client.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{Filters: awsSecurityGroupFilters(infraID)})
+	if defaultSG == nil {
+		defaultSG, err = createAWSSecurityGroup(ctx, r.ec2Client, hcp.Spec.InfraID, vpcID, awsDefaultSecurityGroupName(hcp.Spec.InfraID), "default worker security group", hcp.Spec.Platform.AWS.ResourceTags)
```
Comment (stevekuznetsov) [inline control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:4567]: nit: these functions already take in the infra ID, perhaps the SG name should be a `func(id) string` to ensure the expected relationship is invariant?

**Comment by csrwng** [comment]
Superseded by https://github.com/openshift/hypershift/pull/4742
/close

**Pull request closed without merging**